### PR TITLE
GuidLayoutRenderer - Unique guid per LogEvent

### LINKED ISSUE #60: Add a Unique ID per log event visible by a Layout Renderer
<p>When logging to various targets there isn&#39;t a specific way (that I&quot;m aware of) to link the log events together. The ${guid} layout renderer generates a unique id for each layout. So could we get a single unique reference id that all targets could expose?</p>

<p>&nbsp;</p>

<p>I&#39;d be happy to look into an implementation of this and provide a patch if needed.</p>

### PR DESCRIPTION
**Pull request by snakefoot**
LogEvent will output the same Guid for all Targets. Resolves #60 (And maybe #107)

### CODE REVIEW

File: src/NLog/LayoutRenderers/GuidLayoutRenderer.cs [hunk 1]
```diff
@@ -59,13 +59,34 @@ public GuidLayoutRenderer()
...
+            {
+                byte d = (byte)(LogEventInfo.ZeroDate.Ticks >> 56);
+                byte e = (byte)(LogEventInfo.ZeroDate.Ticks >> 48);
+                byte f = (byte)(LogEventInfo.ZeroDate.Ticks >> 40);
+                byte g = (byte)(LogEventInfo.ZeroDate.Ticks >> 32);
+                byte h = (byte)(LogEventInfo.ZeroDate.Ticks >> 24);
+                byte i = (byte)(LogEventInfo.ZeroDate.Ticks >> 16);
+                byte j = (byte)(LogEventInfo.ZeroDate.Ticks >> 8);
+                byte k = (byte)(LogEventInfo.ZeroDate.Ticks);
+                builder.Append(new Guid(logEvent.SequenceID, (short)logEvent.TimeStamp.Month, (short)logEvent.TimeStamp.Year, d, e, f, g, h, i, j, k).ToString(this.Format));
```
Comment (304NotModified) [inline src/NLog/LayoutRenderers/GuidLayoutRenderer.cs]: Clever approach! 

But is this really a guid? (Global, unique)

Comment (snakefoot) [inline src/NLog/LayoutRenderers/GuidLayoutRenderer.cs]: It is global unique identifier of a LogEvent within the running application. But probably not global in the universe.

File: src/NLog/LayoutRenderers/GuidLayoutRenderer.cs [hunk 2]
```diff
@@ -59,13 +59,38 @@ public GuidLayoutRenderer()
         public string Format { get; set; }
 
         /// <summary>
+        /// Generate the Guid from the NLog LogEvent (Will be the same for all targets)
+        /// </summary>
+        [DefaultValue(false)]
+        public bool LogEventGuid { get; set; }
```
Comment (304NotModified) [inline src/NLog/LayoutRenderers/GuidLayoutRenderer.cs]: What about "basedOnLogEvent"? Or "GeneratedFromLogEvent"

Comment (304NotModified) [reply src/NLog/LayoutRenderers/GuidLayoutRenderer.cs]: @snakefoot ? Your opinion please :)

Comment (snakefoot) [reply src/NLog/LayoutRenderers/GuidLayoutRenderer.cs]: Changed to GeneratedFromLogEvent

**Pull request merged**